app/controllers/collection_show_controller.rb

```diff
@@ -69,9 +69,16 @@ def facet
 
   private
 
-  # Our custom SearchBuilder needs to know collection id (UUID)
+  # Our custom SearchBuilder needs to know:
+  #   collection id (UUID)
+  #   the default sort order for this collection, if specified.
   def search_service_context
-    super.merge!(collection_id: collection.id)
+    super.merge!(collection_id: collection.id, collection_default_sort_order: collection_default_sort_order)
+  end
+
+  # Some collections define a default sort field. Look up its sort order in blacklight_config and use that.
+  def collection_default_sort_order
+    blacklight_config.sort_fields.dig(collection&.default_sort_field)&.sort
```

Comment:
Can we make sure that if the thing in collection.default_sort_field does not exist at all in the blacklight_config -- this doesn't error, and does give us the same default behavior as if it were blank?  It may already do that! Maybe just add a test?

Other than that, looks great!  Good job figuring out the pretty baroque Blacklight architecture stuff!

Reply:
Oh, the collection model validation rules will actually _prevent_ you from saving a value for `collection.default_sort_field` that's not one of the 2 allowed values, namely:
-  `nil`
- `'oldest_date'`.
If you were to remove the 'oldest_date' sort field from catalog_controller, a whole bunch of tests would fail in addition to this one.
We could make this more sophisticated, of course, but for now crude is good.

Reply:
I guess those validation rules are maybe in the next PR i haven't looked at yet, not here? But great!

But one of the cases I'm thinking of is where we _change_ the allowed value, and a value _already_ exists in the db that accidentally is no longer allowed. 

Just seems fairly easy and wise to make sure if this happens it's treated as "no value chosen" rather than blowing up!

Reply:
You are correct - the validation rules are *in fact* not in this PR.

Reply:
OK - added the test!